src/Stratis.Bitcoin/P2P/PeerDiscoveryLoop.cs

```diff
@@ -82,16 +82,18 @@ public sealed class PeerDiscovery : IPeerDiscovery
         }
 
         /// <inheritdoc/>
-        public void DiscoverPeers(NetworkPeerConnectionParameters parameters)
+        public void DiscoverPeers(IConnectionManager connectionManager)
         {
             // If peers are specified in the -connect arg then discovery does not happen.
             if (this.nodeSettings.ConnectionManager.Connect.Any())
                 return;
 
-            if (!parameters.PeerAddressManagerBehaviour().Mode.HasFlag(PeerAddressManagerBehaviourMode.Discover))
+            if (!connectionManager.Parameters.PeerAddressManagerBehaviour().Mode.HasFlag(PeerAddressManagerBehaviourMode.Discover))
```

Comment:
Doesn't the previous line obsolete this line?

Reply:
How do you mean @dangershony ? First we check to see of there are Connect nodes specified at startup and then, if not, we check to see if ```PeerAddressManagerBehaviour``` is set to ```Discover```?

I suppose we can combine that if statement?

WDYT

Reply:
No I mean if there are `connect` nodes then the behaviour will never be set to `discover` right?

Reply:
Ah the `PeerAddressManagerBehaviour` is global so the flag is set once ok thats fine